### ISSUE TITLE
Cannot format doctest with escaped string

### ISSUE THREAD

**Issue by samuelmanzanera** [angelikatyborska/doctest_formatter#14]
Hello

I faced other issue for doctest when I tried to test for example JSON content where the content is string escaped.

Imagine this doctest:
```elixir
@doc """

       iex> %{
       ...>   data: "{\\"supply\\": 100}"
       ...> }
"""
```

This is valid code and valid doctest, however when I ran the formatter I got this issue:
```
** (SyntaxError) syntax error before: supply
     data: "{\\"supply\\": 100}"
               ^
    (elixir 1.14.1) lib/code.ex:773: Code.format_string!/2
    lib/doctest_formatter/formatter.ex:88: DoctestFormatter.Formatter.format_lines/2
    (elixir 1.14.1) lib/enum.ex:4249: Enum.flat_map_list/2
    (elixir 1.14.1) lib/enum.ex:4250: Enum.flat_map_list/2
    lib/doctest_formatter/formatter.ex:61: DoctestFormatter.Formatter.format_doc_content/2
    lib/doctest_formatter/formatter.ex:27: anonymous fn/2 in DoctestFormatter.Formatter.format/2
    (elixir 1.14.1) lib/macro.ex:667: anonymous fn/3 in Macro.prewalk/2
    ```

**Comment by angelikatyborska** [comment]
Thank you for reporting 🙏 you're right, something is wrong in the doctest formatter. I'll take a closer look later today or tomorrow.

**Comment by angelikatyborska** [comment]
Fixing this is not so easy. I will continue trying the next time I have some free time (in a week or so).

I think there's also a temporary workaround. You could rewrite your test not to use interpolated strings with the S sigil:

```elixir
@doc ~S"""

       iex> %{
       ...>   data: "{\"supply\": 100}"
       ...> }
"""
```

**Comment by samuelmanzanera** [comment]
Thanks it does the trick

**Comment by angelikatyborska** [comment]
I made a few more attempts at fixing this and ultimately failed 😞 

I published version 0.3.0 just now in which this kind of problem won't cause the whole formatting process to crash. Instead, a warning will be printed informing the user about the problem and linking to the README and the list of known issues with this library. The one doctest would then remain unformatted, but all the other doctests that don't have this problem would still get formatted. I think that's a reasonable compromise.
